### ISSUE TITLE
GitLab 9.0 API Changes

### ISSUE THREAD

**Issue by connorshea** [NARKOZ/gitlab#235]
This is a "meta" tracking issue for the API changes that will effect the gitlab gem:

- #233
- #229
- #222

See also:
- [Our GitLab CE issue for tracking all changes we intend to make to the API with 9.0.](https://gitlab.com/gitlab-org/gitlab-ce/issues/20070)

**Comment by mltsy** [comment]
I guess that issue I just linked is listed in the [tracking ticket on Gitlab CE](https://gitlab.com/gitlab-org/gitlab-ce/issues/20070) now that I look.  Would you prefer only changes that aren't listed there (I assume they all would be), or do you plan to create issues for each of these changes?

**Comment by asedge** [comment]
@connorshea All the linked issues have been closed.  I intend on releasing `4.0.0` "soon".  Are there any additional APIv4 issues we should tackle before that happens? Personally, I'm OK with releasing `4.0.0` and fixing additional compatibility issues as they arise.

**Comment by asedge** [comment]
`4.0.0` is released.  I don't think all APIv4 route changes have been updated in the gem but hopefully the community will step up and start filing PRs once they encounter issues.

**Comment by connorshea** [comment]
@asedge awesome, thanks :D

**Comment by edgemaster** [comment]
I note that Build Triggers have been renamed Pipeline Triggers in the GitLab docs, should a similar change/alias be made to module names here?

**Comment by mltsy** [comment]
Might as well put in a new PR for that if you are inclined to (or else a new issue).  It doesn't affect the Client interface at all (they are just called triggers in both the API and the Client), but for documentation purposes it would be nicer for those to match up.

**Comment by k0nserv** [comment]
Is there aything that we know now needs to be done for v4 or will issues arise when it starts getting used? I'd be happy to resolve issues otherwise

**Comment by logicminds** [comment]
The fork project method is broken due to the API change.  Should I make a PR and maintain backwards compatibility with v3?

**Comment by asedge** [comment]
@logicminds Yes please make a PR. We've dropped support for APIv3 as of 4.0.0 so no to backwards compatability.

**Comment by logicminds** [comment]
I noticed the url says api version 'v3'  shouldn't this be on v4?  Also how do I switch to v4?

**Comment by mltsy** [comment]
Oops! Where does it say v3? (link?)

**Comment by logicminds** [comment]
This was the error message: `404 Not Found. Request URI: http://web/api/v3/projects/devops/apache/fork`

**Comment by logicminds** [comment]
nevermind I forgot to change my endpoint to v4
